### PR TITLE
added build.sh file that creates dynamic modules for 10 platforms.

### PR DESCRIPTION
**Pull request by anestos**
To run this do  `./build.sh`

It will build the module in 10 platforms and the result is 10 .so files in the build folder.
The platforms are :

- ubuntu18
- ubuntu20 
- centos6
- centos7 
- centos8
- debian9
- debian10
- Amazon Linux 
- Amazon Linus 2 
- alpine

For a different version of NGINX, do `NGINX_VERSION=X.X.X ./build.sh`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+WORKDIR /tmp
+RUN wget https://ftp.pcre.org/pub/pcre/pcre-8.44.tar.gz && tar xzvf pcre-8.44.tar.gz
+RUN wget https://www.zlib.net/zlib-1.2.11.tar.gz && tar xzvf zlib-1.2.11.tar.gz
+RUN CONFIG_OPTS="--with-pcre=../pcre-8.44 --with-zlib=../zlib-1.2.11" ./configure && make
+
+######
+FROM centos:6 as centos6-builder
+
+RUN yum install -y \
+ gcc pcre-devel zlib-devel
```
Comment (travisspencer) [inline Dockerfile:39]: different formatting then above and extra new line below

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+COPY config /tmp
+COPY Makefile /tmp
+COPY phantom_token.c /tmp
+ARG NGINX_VERSION
+ADD nginx-$NGINX_VERSION.tar.gz /tmp/
+
+WORKDIR /tmp
+RUN ./configure && make
+
+
```
Comment (travisspencer) [inline Dockerfile:169]: also extra space

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+COPY --from=centos8-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/centos8.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+COPY --from=debian9-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/debian9.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+COPY --from=debian10-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/debian10.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+COPY --from=amzn-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/amzn.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+COPY --from=amzn2-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/amzn2.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+COPY --from=alpine-builder /tmp/nginx-$NGINX_VERSION/objs/ngx_curity_http_phantom_token_module.so /build/alpine.ngx_curity_http_phantom_token_module.$NGINX_VERSION.so
+
+ENTRYPOINT ["sleep"]
+
+CMD ["300"]
```
Comment (travisspencer) [inline Dockerfile:187]: missing new line

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+NGINX_VERSION=${NGINX_VERSION:-1.19.0}
+NGINX_TARBALL=nginx-${NGINX_VERSION}.tar.gz
+
+if [[ ! -r $NGINX_TARBALL ]]; then
+    wget https://nginx.org/download/nginx-"${NGINX_VERSION}".tar.gz
```
Comment (travisspencer) [inline build.sh:7]: curl is more likely to exist on peoples' machines. Would suggest using that.

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  --build-arg NGINX_SRC_DIR=/tmp/nginx-"$NGINX_VERSION" \
+  --build-arg NGINX_VERSION="$NGINX_VERSION" \
+  --build-arg NGINX_DEBUG=n \
+  --build-arg DYNAMIC_MODULE=Y \
+  -f Dockerfile .
+
+docker run --name nginx-modules -d nginx-module-builder 300
+docker cp nginx-modules:/build/ .
+docker stop -t 0 nginx-modules
+docker rm nginx-modules
```
Comment (travisspencer) [inline build.sh:20]: new line

**Review by travisspencer — APPROVED** [review]
VERY good. This is what we always needed. Great work. Do we already say in the readme that Docker is required for building? If not, we should add that. Trivial, insignificant changes suggested, but not a problem if you don't make them.

**Comment by anestos** [comment]
> VERY good. This is what we always needed. Great work. Do we already say in the readme that Docker is required for building? If not, we should add that. Trivial, insignificant changes suggested, but not a problem if you don't make them.

I didn't add anything in the readme regarding building with this script. I ll do that

**Pull request merged**